# Neural Network gives a Predicition

In [1]:
import numpy as np

w13=0.06
w14=-0.05
w15=0.07

w23=0.06
w24=-0.01
w25=-0.06

w36=0.05
w46=-0.04
w56=0.03

w37=0.02
w47=0.01
w57=0.09

b3=0
b4=0
b5=0

b6=0
b7=0

# Layer 1 Weights

In [2]:
WL1=np.array([[w13,w23],
             [w14,w24],
              [w15,w25]])

bL1=np.array([b3,b4,b5]).reshape(-1,1)

print(WL1,WL1.shape)

[[ 0.06  0.06]
 [-0.05 -0.01]
 [ 0.07 -0.06]] (3, 2)


# Layer 2 Weights

In [3]:
WL2=np.array([[w36,w46,w56],
             [w37,w47,w57]])

bL2=np.array([b6,b7]).reshape(-1,1)

print(WL2,WL2.shape)

[[ 0.05 -0.04  0.03]
 [ 0.02  0.01  0.09]] (2, 3)


# inputs

In [4]:
x1=37
x2=40

Xs=np.array([x1,x2]).reshape(-1,1)
print(Xs,Xs.shape)

[[37]
 [40]] (2, 1)


# Feed forawarding (Forward Propagation)

In [5]:
#Layer 2

x1=37
x2=40

YL2=np.matmul(WL1,Xs)+bL1
print(YL2)

[[ 4.62]
 [-2.25]
 [ 0.19]]


# Output

In [6]:
#Layer 3 - Output Layer

YL3=np.matmul(WL2,YL2)+bL2
print(YL3)

[[0.3267]
 [0.087 ]]


# Actual vs Predicted

Y6(rAIN)       actual(0)      Predicted (0.3267)
Y7(No rain)    actual(1)      Predicted (0.087)

# Error loss

In [8]:
error=np.power(0-0.33,2)+np.power(1-0.09,2)/2
print(error)

0.52295
